workflow/controller/operator_workflow_template_ref_test.go

```diff
@@ -216,6 +216,114 @@ func TestWorkflowTemplateRefParamMerge(t *testing.T) {
 	})
 }
 
+var wftWithValueFromParam = `
+apiVersion: argoproj.io/v1alpha1
+kind: WorkflowTemplate
+metadata:
+  name: wf-template-echo
+  namespace: argo
+spec:
+  entrypoint: echo
+  arguments:
+    parameters:
+      - name: message
+        valueFrom:
+          configMapKeyRef:
+            name: config-properties
+            key: message
+  templates:
+    - name: echo
+      container:
+        image: busybox
+        command: [echo]
+        args: ["{{workflow.parameters.message}}"]
+`
+
+var wfWithValueParamOverride = `
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: wf-parameter-overwrite-
+  namespace: argo
+spec:
+  entrypoint: echo
+  arguments:
+    parameters:
+      - name: message
+        value: "configmap argument overwrite with argument"
+  workflowTemplateRef:
+    name: wf-template-echo
+`
+
+// https://github.com/argoproj/argo-workflows/issues/14426
+func TestWorkflowTemplateRefValueFromParamOverwrite(t *testing.T) {
+	wf := wfv1.MustUnmarshalWorkflow(wfWithValueParamOverride)
+	wftmpl := wfv1.MustUnmarshalWorkflowTemplate(wftWithValueFromParam)
+	t.Run("CheckArgumentFromWFT", func(t *testing.T) {
+		cancel, controller := newController(wf, wftmpl)
+		defer cancel()
+		ctx := context.Background()
+		woc := newWorkflowOperationCtx(wf, controller)
+		woc.operate(ctx)
+		assert.Equal(t, wf.Spec.Arguments.Parameters, woc.wf.Spec.Arguments.Parameters)
+		assert.Equal(t, "configmap argument overwrite with argument", woc.wf.Spec.Arguments.Parameters[0].Value.String())
+	})
+}
+
+var wftWithValueParameter = `
+apiVersion: argoproj.io/v1alpha1
+kind: WorkflowTemplate
+metadata:
+  name: wf-template-echo
+  namespace: default
+spec:
+  entrypoint: echo
+  arguments:
+    parameters:
+      - name: message
+        value: "message from workflow template"
+  templates:
+    - name: echo
+      container:
+        image: busybox
+        command: [echo]
+        args: ["{{workflow.parameters.message}}"]
+`
+
+var wfWithValueFromParamOverride = `
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: wf-parameter-overwrite-
+  namespace: default
+spec:
+  entrypoint: echo
+  arguments:
+    parameters:
+      - name: message
+        valueFrom:
+          configMapKeyRef:
+            name: config-properties
+            key: message
+  workflowTemplateRef:
+    name: wf-template-echo
+`
+
+func TestWorkflowTemplateRefValueParamOverwrite(t *testing.T) {
```

Comment:
@MasonM this test is showing that the merge logic is not correct, let me know what you think

I found a way to use the ConfigMap, but I am not sure it's needed (I was looking at the tests in `operator_concurrency_test.go` where a`ConfigMap` is used). As it is written, the test shows already that the overwrite is working, but when trying to get the value from it, since the confimap informer is missing, the value is not extracted. The code needed to not fail has been added in a separate commit

Reply:
here is the code for the ConfigMap informer to run: https://github.com/argoproj/argo-workflows/pull/14462/commits/8a5782f63cb9fe2f0b2a0e4bb44e252c63d27ae2